### PR TITLE
WIP: Add some new test scripts

### PR DESCRIPTION
**Pull request by xuhang57**
## Description

Adding a new test script for `api/v2/project_instances/##`.
Adding a new test script for `api/v2/sizes/`
Adding a new test script for `api/v2/instances/##/actions`

## Checklist before merging Pull Requests
- [x] New test(s) included to reproduce the bug/verify the feature
- [ ] Documentation created/updated at [Example link to documentation](https://example.test/doc#new_section) to give context to the feature
- [ ] Reviewed and approved by at least one other contributor.
- [ ] If necessary, include a snippet in CHANGELOG.md

### CODE REVIEW

**Comment by xuhang57** [comment]
Having 
```
ProtocolError: ("Connection broken: error(104, 'Connection reset by peer')", error(104, 'Connection reset by peer'))```

**Comment by xuhang57** [comment]
@steve-gregory  `resize` test fails due to we don't have the `resize` refactor here along with this PR. I tested in my local server and it works.

**Comment by xuhang57** [comment]
Curious how I could write Documentation for this? Thanks!

**Comment by xuhang57** [comment]
no longer a WIP. free feel to review it  👍

File: api/tests/factories/leadership_factory.py [hunk 1]
```diff
@@ -4,5 +4,15 @@
...
 class LeadershipFactory(factory.DjangoModelFactory):
 
+    @classmethod
+    def _create(cls, model_class, *args, **kwargs):
+        leadership = model_class(*args, **kwargs)
+        leadership.save()
+        # This happens by default in Atmosphere, but is not reflected in the LeadershipFactory alone.
+        user = leadership.user
+        group = leadership.group
+        group.user_set.add(user)
```
Comment (steve-gregory) [inline api/tests/factories/leadership_factory.py]: 👍

File: api/tests/v2/test_instance_actions.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        data = {
+            "action":"resize",
+            "resize_size":self.size_large.alias
+        }
+        request = factory.post(self.url, data)
+        force_authenticate(request, user=self.user)
+        response = self.view(request, str(self.active_instance.provider_alias))
+        self.assertEquals(response.status_code, 200)
+        data = response.data.get('result')
+        self.assertEquals('success', data)
```
Comment (steve-gregory) [inline api/tests/v2/test_instance_actions.py]: 👍

**Comment by xuhang57** [comment]
Thanks for the great help 👍

**Pull request merged**